tests/e2e/test_launch.py

```
@@ -105,6 +105,29 @@ def test_package_import():
     return test_helper.test_package_import()
 
 
+def test_chain_integration_import():
+    """Test that chain integration imports work correctly."""
+    test_helper = StreamlitE2ETest()
+
+    # Test specific chain-related imports
+    result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)